core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -56,10 +59,16 @@ public abstract class ViewCatalogTests<C extends ViewCatalog & SupportsNamespace
 
   protected abstract Catalog tableCatalog();
 
+  @TempDir private Path tempDir;
```

Comment:
previously we were referring directly to the `tmp` dir (similar to `CatalogTests`), but it's better to use a randomly generated directory. I'll have a follow-up PR to also use a randomly generated temp dir for `CatalogTests`